lib/knock/resources/objects.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Knock.Objects do
+  @moduledoc """
+  Knock resources for accessing Objects
+  """
+  alias Knock.Api
+  alias Knock.Client
+
+  @typedoc """
+  An object reference is how we refer to a particular object in a collection
+  """
+  @type ref :: %{id: :string, collection: :string}
+
+  @doc """
+  Builds an object reference, which can be used in workflow trigger calls.
+  """
+  def build_ref(collection, id), do: %{id: id, collection: collection}
+
+  @doc """
+  Sets the given object, performing an upsert operation in the process.
+  """
+  def set(client, collection, id, attrs) do
+    Api.put(client, "/objects/#{collection}/#{id}", attrs)
+  end
+
+  @doc """
+  Gets the given object.
+  """
+  def get(client, collection, id) do
+    Api.get(client, "/objects/#{collection}/#{id}")
+  end
+
+  @doc """
+  Soft deletes the given object.
+  """
+  def delete(client, collection, id) do
+    Api.delete(client, "/objects/#{collection}/#{id}")
+  end
+
+  ##
+  # Bulk functions
+  ##
+
+  @doc """
+  Bulk sets one or more objects in a collection
+  """
+  def bulk_set(client, collection, objects_to_bulk_set) do
+    Api.post(client, "/objects/#{collection}/bulk/set", %{objects: objects_to_bulk_set})
+  end
+
+  @doc """
+  Bulk deletes one or more objects in a collection
```

Comment:
Nit: Noticing we conclude with a period on the other comments.

```suggestion
  Bulk deletes one or more objects in a collection.
```